src/views/TrackPositionAnimation/TPATimeSyncLogic.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useCallback, useMemo } from "react"
+import useBinarySearchTree, { BstSearchFn, BstSearchResult } from "util/bst"
+import { AnimationState, AnimationStateAction } from "views/common/Animation/AnimationStateReducer"
+import { PositionFrame } from "./TrackPositionAnimationTypes"
```

Comment:
This file provides three basic bits of functionality:
- Given the time-series data, it constructs a memoized binary search tree of the times & the corresponding frames, and returns a search function generator that makes it easy to find the frame closest to any given time code
- A `matchFocusToFrame` function that updates the focus time given the animation state (which identifies the current frame) if the current frame's time is more than a minimum distance from the focus time. This is expected to be called when the animation playback mode changes (but could handle other calling conditions too)
- A `matchFrameToFocus` function that should be called in a `useEffect` hook driven by the focus time; this sets the animation playback current frame to the one closest to the focus time when the focus time changes. (The hook should be set up so this triggers only when the change is external to the track position animation view.)